server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java

```diff
@@ -371,6 +371,9 @@ private static long getTotalDiskUsageInBytes(ClusterInfo clusterInfo, Metadata m
 
         // Visible for testing
         static long getIndexDiskUsageInBytes(ClusterInfo clusterInfo, IndexMetadata indexMetadata) {
+            if (indexMetadata.ignoreDiskWatermarks()) {
```

Comment:
Can we add a comment that mentions that we ignore disk watermarks for partially mounted/frozen indices?